cpp/tests/merge/merge-test.cu

```diff
@@ -0,0 +1,466 @@
+    auto expectedDataWrap2 = columnFactory.make(outputRows,
+                                                [](gdf_index_type row)->gdf_index_type {
+                                                    if(cudf::gdf_dtype_of<TypeParam>() == GDF_BOOL8) return 0;
+                                                    else return row / 2; 
+                                                });
+
+    EXPECT_TRUE(gdf_equal_columns(*expectedDataWrap1.get(), *outputTable.get_column(0)));
+    EXPECT_TRUE(gdf_equal_columns(*expectedDataWrap2.get(), *outputTable.get_column(1)));
+}
+
+TYPED_TEST(MergeTest, Merge2KeyColumns) {
+    cudf::test::column_wrapper_factory<TypeParam> columnFactory;
+
+    gdf_size_type inputRows = 50000;
+    inputRows = (cudf::detail::unwrap(std::numeric_limits<TypeParam>::max()) < inputRows ? 40 : inputRows);
+
+    auto leftColWrap1 = columnFactory.make(inputRows,
+                                            [=](gdf_index_type row)->gdf_index_type {
+                                                if(cudf::gdf_dtype_of<TypeParam>() == GDF_BOOL8) return row >= inputRows / 2;
+                                                else return row;
+                                            });
+    auto leftColWrap2 = columnFactory.make(inputRows,
+                                            [=](gdf_index_type row)->gdf_index_type {
+                                                if(cudf::gdf_dtype_of<TypeParam>() == GDF_BOOL8) return (row / (inputRows / 4)) % 2 == 0;
+                                                else return 2 * row;
+                                            });
+    gdf_column *leftColumn1 = leftColWrap1.get();
+    gdf_column *leftColumn2 = leftColWrap2.get();
+
+    auto rightColWrap1 = columnFactory.make(inputRows,
+                                            [=](gdf_index_type row)->gdf_index_type {
+                                                if(cudf::gdf_dtype_of<TypeParam>() == GDF_BOOL8) return row >= inputRows / 2;
+                                                else return row;
+                                            });
+    auto rightColWrap2 = columnFactory.make(inputRows,
+                                            [=](gdf_index_type row)->gdf_index_type {
+                                                if(cudf::gdf_dtype_of<TypeParam>() == GDF_BOOL8) return (row / (inputRows / 4)) % 2 == 0;
+                                                else return 2 * row + 1;
+                                            });
+    gdf_column *rightColumn1 = rightColWrap1.get();
+    gdf_column *rightColumn2 = rightColWrap2.get();
+
+    gdf_column *leftColumns[]  = {leftColumn1, leftColumn2};
+    gdf_column *rightColumns[] = {rightColumn1, rightColumn2};
+
+    std::vector<gdf_size_type> sortByCols = {0, 1};
+    std::vector<order_by_type> orderByTypes = {GDF_ORDER_ASC, GDF_ORDER_DESC};
+
+    cudf::table outputTable;
+    EXPECT_NO_THROW(outputTable = cudf::merge(cudf::table(leftColumns, 2),
```

Comment:
Just FYI, `cudf::table` has a `std::initializer_list<gdf_column*>` constructor that can simplify your code here a bit. 
This way you don't need to make an explicit array of `gdf_column*` in `leftColumns[]` and `rightColumns[]` 

```suggestion
    EXPECT_NO_THROW(outputTable = cudf::merge(cudf::table({leftColWrap1.get(), leftColWrap2.get()}),
```